tests/test_counter.py

```diff
@@ -0,0 +1,18 @@
+"""This module tests the castanet.counter module."""
+
+from castanet import counter
+from castanet import generate_trees as generator
+
```

Comment:
You should have test cases determining that the `while_dictionary` and `for_dictionary` contain the proper things. 

There should be additional test cases that test if your total function is working properly.